### PR TITLE
Add github action file for stylelint checks on only diffs for PRs

### PR DESCRIPTION
**Pull request by matthewroach**
## Description
Adds a GitHub action for running stylelint against the code changed in the PR. 

* Its set to only run on the diff on each commit tolerance of the diff check is -+ 3 lines

## Jira Ticket
- [PEN-1694](https://arcpublishing.atlassian.net/browse/PEN-1694)

## Test Steps

Commit a change to some Sass/CSS that would trigger a failure

## Effect Of Changes

See PR #708 for example of GitHub action in "action" 😄 

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [ ] Confirmed all the test steps a reviewer will follow above are working. 
- [ ] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [ ] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [ ] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
- [ ] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist 
_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr. 
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

File: .github/workflows/stylelint-pr.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+      - name: stylelint
+        uses: reviewdog/action-stylelint@v1
+        with:
+          github_token: ${{ secrets.github_token }}
+          fail_on_error: true
+          reporter: github-pr-review
+          filter_mode: 'diff_context'
+          stylelint_config: '.stylelintrc.json'
+          stylelint_input: '**/*.(scss|jsx)'
```
Comment (JackHowa) [inline .github/workflows/stylelint-pr.yml:47]: this should probably be just .scss. have had issues with jsx locally

File: .github/workflows/stylelint-pr.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
+name: Stylelint PR Diffs
```
Comment (JackHowa) [inline .github/workflows/stylelint-pr.yml:1]: suggestion: Lint scss styles 

Something just a bit more plain english readable. Just a suggestion

**Review by JackHowa — APPROVED** [review]
great work! will keep jsx in there as a suggestion in case styled components and stylelint don't play nice

File: .github/workflows/stylelint-pr.yml (hunk 2 above)
Comment (matthewroach) [reply .github/workflows/stylelint-pr.yml:1]: As discussed in slack. Going leave this as is.

Ideally we should call it this - but that can be used for when we have all lint errors down and we are running it all the time and not just on the PR diff set

**Pull request merged**